chatkit-core/src/integrationTest/kotlin/com/pusher/chatkit/RoomSpek.kt

```diff
@@ -83,32 +85,26 @@ object RoomSpek : Spek({
             assertThat(updatedRoom.name).isEqualTo(NOT_GENERAL)
         }
 
-        it("notifies '$ALICE' when room '$GENERAL' receives a new message") {
-            setUpInstanceWith(createDefaultRole(), newUsers(ALICE, PUSHERINO), newRoom(GENERAL, ALICE, PUSHERINO))
+        it("updates the unread count and last message at when '$GENERAL' receives a new message") {
```

Comment:
this test is still changed, whilst the [intention ](https://github.com/pusher/chatkit-android/pull/148#discussion_r370208336) was to revert the changes

Reply:
I think I may have misunderstood - i thought we wanted to combine the two tests - are you asking my to revert the change that test and keeping what we had previously? If so, we can just abandon this PR?

Reply:
The test used to combine the two checks (lastMessageAt and unreadCount) and that's needed as we receive 2 RoomUpdated events for each property and don't have guarantee on the order of them.

Now the test is just checking lastMessageAt event and not as it used to i.e. not from the perspective of the message recipient (which is more important than the sender's perspective). Also it uses junit assertions as opposed to Truth.

Let's revert the changes in this test.

The extraction and fixing of flakiness in custom data test is definitely a nice thing, not related to the ME effort but as it got discovered as part of it, I would be keen on getting it merged.

Reply:
Okay, i'll update the title of this PR to improve clarity around what's actually happening now.